src/hooks/lib/preview/useCurvePreviewDepositUnderlying.ts

```diff
@@ -0,0 +1,105 @@
+import { Address, useContractRead, useContractReads } from "wagmi"
+
+import { VaultType } from "@/lib/types"
+import { useVaultContract } from "@/hooks/lib/useVaultContract"
+import { useActiveChainId } from "@/hooks/useActiveChainId"
+
+import { CurvePool2Assets, CurvePool3Assets } from "@/constant/abi"
+import {
+  ARBI_CURVE_ADDRESS,
+  crvTriCryptoPoolAddress,
+  ETH,
+} from "@/constant/addresses"
+
+const WETH_ARBI: Address = "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1"
+const WETH: Address = "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"
+
+export default function useCurvePreviewDeposit({
+  asset,
+  vaultAddress,
+  token,
+  amount,
+  type,
+  slippage,
+  enabled,
+}: {
+  asset: Address
+  vaultAddress: Address
+  token?: Address
+  amount: string
+  type: VaultType
+  slippage: number
+  enabled?: boolean
+}) {
+  const chainId = useActiveChainId()
+  const isArbitrumFamily = chainId === 313371 || chainId === 42161
+  const poolCurveAddress = ARBI_CURVE_ADDRESS[asset] ?? "0x"
+  token =
+    token === ETH && isArbitrumFamily
+      ? WETH_ARBI
+      : token === ETH && !isArbitrumFamily
+      ? WETH
+      : token
+
+  const underlyingAssets = useContractReads({
+    contracts: [0, 1, 2, 3, 4].map((index) => ({
+      address: poolCurveAddress,
+      chainId,
+      abi: CurvePool3Assets,
+      functionName: "coins",
+      args: [index],
+    })),
+    enabled: poolCurveAddress !== "0x" && enabled,
+    select: (results) =>
+      results.map((item) => {
+        return !item.error ? (item.result as unknown as Address) : "0x"
```

Comment:
```ts
select: results => results.map(item => !item.error ? item.result : "0x")
```

Sidenote: I don't think you need type assertion here. It should already know it's an `Address`